File: app/src/main/java/org/magnum/logger/communication/calls/CallTable.java
package org.magnum.logger.communication.calls;

import android.content.ContentValues;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import org.magnum.logger.Database;

public class CallTable extends Database
{
	private static final String TABLE_NAME = "Calls";
	private static final String[] COLUMN_NAMES = {"Time", "CallID", "Incoming", "Duration"};

	CallTable(Context context)
	{
		super(context);
	}

	public static void create(SQLiteDatabase db)
	{
		db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_NAME + "( Time INTEGER NOT NULL,  CallID INTEGER NOT NULL, Incoming INTEGER NOT NULL, Duration INTEGER )");
	}

	@Override
	public String[] getColumnNames()
	{
		return COLUMN_NAMES;
	}

	@Override
	public String getTableName()
	{
		return TABLE_NAME;
	}

	public void setDuration(long callTime, long duration)
	{
		ContentValues dataToInsert = new ContentValues();                          
		dataToInsert.put("Duration", duration);
		String where = "Time=?";
		String[] whereArgs = new String[] {String.valueOf(callTime)};
		
		SQLiteDatabase db = getWritableDatabase();
		db.update(TABLE_NAME, dataToInsert, where, whereArgs);
		db.close();
	}
}

File: app/src/main/java/org/magnum/logger/communication/calls/CallTable.kt
package org.magnum.logger.communication.calls

import androidx.room.*

const val NAME = "calls"

@Entity(tableName = NAME)
data class CallRecord(@PrimaryKey val time: Long, val callId: String, val isIncoming: Boolean, val duration: Long?)

@Dao
interface CallTable {
    @Insert
    fun insert(vararg record: CallRecord)

    @Query("Update $NAME set duration = :duration where time = :time")
    fun setDuration(time: Long, duration: Long)
}